nomos-services/mempool/src/network/adapters/libp2p.rs

```diff
@@ -24,15 +30,39 @@ where
     type Tx = Tx;
 
     async fn new(
-        _network_relay: OutboundRelay<<NetworkService<Self::Backend> as ServiceData>::Message>,
+        network_relay: OutboundRelay<<NetworkService<Self::Backend> as ServiceData>::Message>,
     ) -> Self {
+        network_relay
+            .send(NetworkMsg::Process(Command::Subscribe(
+                CARNOT_TX_TOPIC.to_string(),
+            )))
+            .await
+            .expect("Network backend should be ready");
         Self {
-            _network_relay,
+            network_relay,
             _tx: PhantomData,
         }
     }
     async fn transactions_stream(&self) -> Box<dyn Stream<Item = Self::Tx> + Unpin + Send> {
-        // TODO
-        Box::new(futures::stream::empty())
+        let topic_hash = TopicHash::from_raw(CARNOT_TX_TOPIC);
+        let (sender, receiver) = tokio::sync::oneshot::channel();
+        self.network_relay
+            .send(NetworkMsg::Subscribe {
+                kind: EventKind::Message,
+                sender,
+            })
+            .await
+            .expect("Network backend should be ready");
+        let receiver = receiver.await.unwrap();
+        Box::new(Box::pin(BroadcastStream::new(receiver).filter_map(
+            move |message| match message {
+                Ok(Event::Message(Message { data, topic, .. })) if topic == topic_hash => {
+                    let tx = wire::deserialize::<TransactionMsg<Tx>>(&data)
+                        .expect("Failed to deserialize tx");
```

Comment:
we probably don't want to panic on this

Reply:
Oh yes, I left for placeholder as I was having some other issue, and forgot to remove. Thanks!!